bellatrix.core/src/main/java/solutions/bellatrix/core/utilities/SingletonFactory.java

```diff
@@ -37,6 +37,7 @@ public static <T> T getInstance(Class<T> classOf, Object... initargs) {
             return (T)SINGLETON_FACTORY.mapHolder.get(classOf.getName());
         } catch (Exception e) {
             // not the best practice to return null. But probably we will never end here so it is OK.
+            Log.error("Failed to create instance of the page. Exception was: " + e);
```

Comment:
Shouldn't the message be more generic? Maybe get the name of the class we tried to get an instance of and use String.format